src/nanoarrow/common/inline_types.h

```diff
@@ -916,13 +922,14 @@ static inline void ArrowDecimalInit(struct ArrowDecimal* decimal, int32_t bitwid
   memset(decimal->words, 0, sizeof(decimal->words));
   decimal->precision = precision;
   decimal->scale = scale;
+  // n_words will be 0 for bitwidth == 32
```

Comment:
I can definitely add an explanation in a comment for it, but I don't know how to avoid the `n_words == 0` situation without updating words to be smaller, personally. I can do that if you want, but I agree it would be hard given how often it is used.

Reply:
Arrow C++ still operates against 64 bit words with decimals right? Guessing the 32 bit implementation upstream still needs to be worked out?

https://github.com/apache/arrow/blob/cd3321b28b0c9703e5d7105d6146c1270bbadd7f/cpp/src/arrow/util/decimal.cc#L527

Reply:
Added a note for the documentation of both `n_words` and `words` to indicate the behavior for 32-bit decimal values.